### PR TITLE
Pivotal ID # 187815483: Async Web Client

### PR DESCRIPTION
**Pull request by Juan-EBI**
https://www.pivotaltracker.com/story/show/187815483
- refactor file api

### CODE REVIEW

File: client/cluster-client/src/main/kotlin/uk/ac/ebi/biostd/client/cluster/api/SlurmClusterClient.kt [hunk 1]
```diff
@@ -97,13 +97,18 @@ class SlurmClusterClient(
...
-        return failure(JobSubmitFailException(response))
+        logger.error { "Error submission job, exitCode='$exitCode', response='$response'" }
+        return failure(JobSubmitFailException(exitCode, response))
     }
 
     companion object {
+        /**
+         * Empirically it was found that both -1, 0 means job was submitted sucefully.
+         */
+        val SUCCESS_EXITS_CODE = setOf(-1, 0)
```
Comment (Juan-EBI) [inline client/cluster-client/src/main/kotlin/uk/ac/ebi/biostd/client/cluster/api/SlurmClusterClient.kt:110]: @jhoanmanuelms noticed this and solution seems to be working. (Could not find anything in regards -1 in the documentation)

File: submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/common/config/SchedulingConfig.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import ac.uk.ebi.biostd.archive.ArchiveScheduler
+import ac.uk.ebi.biostd.common.properties.ApplicationProperties
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionRequestPersistenceService
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.scheduling.annotation.EnableScheduling
+
+@Configuration
+@EnableScheduling
```
Comment (Juan-EBI) [inline submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/common/config/SchedulingConfig.kt:11]: this annotation was missing from schedule jobs to work

File: commons/commons-util/src/main/kotlin/ebi/ac/uk/io/KFiles.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package ebi.ac.uk.io
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.nio.file.Files
+import java.nio.file.Path
+
+object KFiles {
```
Comment (jhoanmanuelms) [inline commons/commons-util/src/main/kotlin/ebi/ac/uk/io/KFiles.kt:8]: 👍

**Review by jhoanmanuelms — APPROVED** [review]
LGTM. `suspend`everywhere!!!

image

**Pull request merged**